vortex-ipc/src/chunked_reader/mod.rs

```diff
@@ -23,17 +20,30 @@ pub struct ChunkedArrayReader<R: VortexReadAt> {
 }
 
 impl<R: VortexReadAt> ChunkedArrayReader<R> {
+    pub fn try_new(
+        read: R,
+        view_context: Arc<ViewContext>,
+        dtype: DType,
+        byte_offsets: Array,
```

Comment:
FLUP work, but we should create a `struct MetadataArray(Array)` that validates the inner array as a struct schema with the expected columns. Can then provide typed accessors for e.g. byte_offsets, or statistics, etc.